test/NVQPP/qir_gen.cpp

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2025 NVIDIA Corporation & Affiliates.                         *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ %cpp_std --emit-qir %s && cat qir_gen.qir.ll | \
```

Comment:
The structure of this test is very broad and less focused on the exact issue. That is, something in a completely unrelated area of the compiler, or even runtime headers, might change this such that the focus of the test vanishes.

Can we instead capture the Quake code that has the arith.select operation and run `cudaq-translate --convert-to=qir %s -o %t` on that? That will guarantee that other changes don't introduce false positive results.

Reply:
Yeah, that's good. I was asking about making the test take a .qke file as input as well. If you want, you can open this comment as a new issue for later. It's a good first issue. :)